### ISSUE TITLE
Upload new file with translations with the API fails

### ISSUE THREAD

**Issue by andonigonzalez** [WeblateOrg/weblate#1476]
### Steps to reproduce
`curl -v -X POST -F file=@file_full_path -H "Authorization:Token TOKEN" http://weblate.local/api/translations/proyect/component/language/file/`

### Actual behaviour
I am trying to upload a new file with translations(.NET resource file) using the API and it returns 
`{"count":29,"accepted":0,"skipped":4,"result":false,"not_found":0,"total":29}`
The file I am trying to upload has 4 strings, and it skipped all of them.

### Expected behaviour
If I use weblate site and upload the same .resx file, it works well and update the strings perfectly.

**Comment by nijel** [comment]
There is currently no way to overwrite existing translations using the API, it just can change non translated ones...

**Comment by andonigonzalez** [comment]
Okay, thank you for answering so quickly

**Comment by nijel** [comment]
Thank you for your report, the issue you have reported has just been fixed. 

* In case you see problem with the fix, please comment on this issue. 
* In case you see similar problem, please open separate issue.
* If you are happy with the outcome, consider supporting Weblate by [donating](https://weblate.org/donate/).

**Comment by andonigonzalez** [comment]
I am uploading new traductions files and it overwrites existing traductions well, but it takes so much time to complete the upload.
This appears in the apache log file

`[Fri May 12 12:57:36.552315 2017] [wsgi:error] [pid 3313] [remote 127.0.0.1:36090] ERROR global-resources-views/error/eu/202277: failed to lock backend for 83b9c11d714baaba on GLOBAL-Resources-Views/Error - Basque!`

And after two or three times, this one

`[Fri May 12 12:57:36.616131 2017] [wsgi:error] [pid 3313] [remote 127.0.0.1:36090] INFO global-resources-views/live-error/eu: commiting`

Finally it complete the upload and the traductions are well, but it takes like 15 minutes to upload all files, I do not know if it is my problem or it is an issue.

Hope you could help me

**Comment by nijel** [comment]
Apparently you're not using API, right? Please open separate issue then as it's not related here...

**Comment by andonigonzalez** [comment]
Yes I'm using API, specifically this method

`POST /api/translations/(string: project)/(string: component)/(string: language)/file/`

I'm using it in a for loop to upload many files and I don't know why it fires that error, but not with all files

**Comment by nijel** [comment]
Are you doing the uploads in parallel? Is there some other activity going on?

**Comment by andonigonzalez** [comment]
No, I'm executing de http request with php exec function:
`exec("curl -F overwrite=true -F file=@$ruta -H 'Authorization: Token TOKEN' $uploadUrl", $res2);`
and it waits until the execution is finished before processing the next line, so there is no upload in parallel.

And there is no other activity, I have a form from I uploads the files, that's it.

**Comment by nijel** [comment]
Ah, now I see the problem (it would happen on file upload using web as well), should be fixed now.

**Comment by andonigonzalez** [comment]
Thank you so much, now it works like a charm!
